test/components/gridlinesTests.ts

```diff
@@ -1,63 +1,116 @@
 
-    let gridlines = new Plottable.Components.Gridlines(xScale, yScale);
     let basicTable = new Plottable.Components.Table().add(yAxis, 0, 0)
                                           .add(gridlines, 0, 1)
                                           .add(xAxis, 1, 1);
 
     basicTable.anchor(svg);
     basicTable.computeLayout();
-    xScale.range([0, xAxis.width() ]); // manually set range since we don't have a renderer
-    yScale.range([yAxis.height(), 0]);
     basicTable.render();
 
-    let xAxisTickMarks = (<any> xAxis)._element.selectAll("." + Plottable.Axis.TICK_MARK_CLASS)[0];
-    let xGridlines = (<any> gridlines)._element.select(".x-gridlines").selectAll("line")[0];
+    const xAxisTickMarks = xAxis.content().selectAll("." + Plottable.Axis.TICK_MARK_CLASS);
+    const xGridlines = gridlines.content().select(".x-gridlines").selectAll("line");
     assert.strictEqual(xAxisTickMarks.length, xGridlines.length, "There is an x gridline for each x tick");
-    for (let i = 0; i < xAxisTickMarks.length; i++) {
-      let xTickMarkRect = xAxisTickMarks[i].getBoundingClientRect();
-      let xGridlineRect = xGridlines[i].getBoundingClientRect();
+    xAxisTickMarks.each(function(tickMark, i) {
+      let xTickMarkRect = this.getBoundingClientRect();
+      let xGridlineRect = (<Element> xGridlines[0][i]).getBoundingClientRect();
       assert.closeTo(xTickMarkRect.left, xGridlineRect.left, 1, "x tick and gridline align");
-    }
+    });
 
-    let yAxisTickMarks = (<any> yAxis)._element.selectAll("." + Plottable.Axis.TICK_MARK_CLASS)[0];
-    let yGridlines = (<any> gridlines)._element.select(".y-gridlines").selectAll("line")[0];
+    let yAxisTickMarks = yAxis.content().selectAll("." + Plottable.Axis.TICK_MARK_CLASS);
+    let yGridlines = gridlines.content().select(".y-gridlines").selectAll("line");
     assert.strictEqual(yAxisTickMarks.length, yGridlines.length, "There is an x gridline for each x tick");
-    for (let j = 0; j < yAxisTickMarks.length; j++) {
-      let yTickMarkRect = yAxisTickMarks[j].getBoundingClientRect();
-      let yGridlineRect = yGridlines[j].getBoundingClientRect();
+
+    yAxisTickMarks.each(function(tickMark, i) {
+      let yTickMarkRect = this.getBoundingClientRect();
+      let yGridlineRect = (<Element> yGridlines[0][i]).getBoundingClientRect();
       assert.closeTo(yTickMarkRect.top, yGridlineRect.top, 1, "y tick and gridline align");
-    }
+    });
 
     svg.remove();
   });
+
+  it("updates gridlines when scales update", () => {
+    gridlines.renderTo(svg);
+
+    let xGridlines = gridlines.content().select(".x-gridlines").selectAll("line");
+    let xTicks = xScale.ticks();
+    assert.strictEqual(xGridlines.size(), xTicks.length, "There is an x gridline for each x tick");
+    xGridlines.each(function(gridline, i) {
+      const x = TestMethods.numAttr(d3.select(this), "x1");
+      assert.closeTo(x, xScale.scale(xTicks[i]), 1, "x gridline drawn on ticks");
```

Comment:
This is the logic that should be in the "uses tick generators" test.

Also, we should use either `strictEqual`, or `closeTo` with `window.Pixel_CloseTo_Requirement`.